src/components/CourseCard/EmailSettingsModal.jsx

```diff
@@ -1,57 +1,138 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {Button, CheckBox, Icon, Modal} from '@edx/paragon';
+import { Input, Modal, StatusAlert } from '@edx/paragon';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 import './EmailSettingsModal.scss';
 
+class EmailSettingsModal extends Component {
+  state = {
+    hasEmailsEnabled: this.props.hasEmailsEnabled || false,
+    isSubmitting: false,
+    isFormChanged: false,
+    error: null,
+  };
 
-const EmailSettingsModal = (props) => {
-  const modalRef = React.createRef();
-  const {
-    title, submitting, onClose, onEmailSettingsChange,
-  } = props;
-
-  return (
-    <React.Fragment>
-      <Modal
-        ref={modalRef}
-        title={title}
-        body={
-          <React.Fragment>
-            <CheckBox
-              name="email_settings"
-              label={
-                <React.Fragment>
-                  <span>
-                    Receive course emails such as reminders, schedule updates,
-                    and other critical announcements.
-                  </span>
-                  &nbsp; &nbsp;
-                  {submitting && <Icon className={['fa', 'fa-spinner', 'fa-spin', 'mr-2']} />}
-                </React.Fragment>
-              }
-              disabled={submitting}
-              onChange={onEmailSettingsChange}
-            />
-          </React.Fragment>
-        }
-        onClose={onClose}
-        closeText={submitting ? <Icon className={['fa', 'fa-spinner', 'fa-spin', 'mr-2']} /> : 'Save Settings'}
-        open
-      />
-    </React.Fragment>
-  );
-};
+  handleEmailSettingsChange = (e) => {
```

Comment:
Sets `isFormChanged` based on whether the newly changed `checked` state differs from the initial `hasEmailsEnabled` prop passed in.